File: randlib/randlib.cs
using System;
using System.Linq;

namespace Randlib
{

	public static class randlib
	{
		static private ulong RANDLIB_MAX = 32767;
		static private ulong t1, t2;
		
		public static void Randinit(){
			t1 = (ulong)DateTime.Now.Ticks;
		}
		
		public static void Set_RANDLIB_MAX(ulong value){
			RANDLIB_MAX = value;
		}
		
		private static int NewRand(){
			uint value;
			t2 = (ulong)DateTime.Now.Ticks;
			ulong seed = (ulong)(t2 - t1);
			value = (uint)((ulong)(8253738 * seed + 2396403) % RANDLIB_MAX);
			return (int)value;
		}
		
		public static int Randint(int min, int max){
			if (min > max){
				(min, max) = (max, min);
			}
			return min + (NewRand() % (max - min + 1));
		}
		
		public static short Randshort(short min, short max){
			if (min > max){
				(min, max) = (max, min);
			}
			return (short)((int)min + (NewRand() % (int)(max - min + 1)));
		}
		
		public static long Randlong(long min, long max){
			if (min > max){
				(min, max) = (max, min);
			}
			return (long)((int)min +(NewRand() % (int)(max - min +1)));
		}
		
		public static float Randfloat(float min, float max){
			if (min > max){
				(min, max) = (max, min);
			}
			long pre;
			float post;
			pre = Randint((int)min, (int)(max-1));
			post = (float)NewRand() / (float)Math.Pow(10, (float) Math.Log10(RANDLIB_MAX));
			return (float)post + (float)pre;
		}
		
		public static double Randdouble(double min, double max){
			if (min > max){
				(min, max) = (max, min);
			}
			long pre;
			double post;
			pre = Randint((int)min, (int)(max-1));
			post = (double)NewRand() / (double)Math.Pow(10, (double)Math.Log10(RANDLIB_MAX));
			return (double)post + (double)pre;
		}
		
		public static bool Randbool(){
			return Randint(0,1) == 1;
		}
		
		public static char Randchar(bool excludeWhitespaces){
			char[] whitespaces = {' ', '\t', '\n'};
			if (excludeWhitespaces){
				return (char)Randint(33, 126);
			}
			else{
				return Randint(0,6) != 0 ? (char)Randint(33, 126) : whitespaces[Randint(0, whitespaces.Length-1)];
			}
		}
		
		public static string Randstring(uint length, bool excludeWhitespaces){
			string text = "";
			for (int i = 0; i < length; i++){
				text += Randchar(excludeWhitespaces);
			}
			return text;
		}
		
		
	}
}

File: randlib/randlib.go
package randlib

import (
	"math"
	"time"
)

var seed uint64 = 0
var RANDLIB_MAX uint64 = 32767
var prevSeed uint64 = 0

func Set_RANDLIB_MAX(value uint64) {
	RANDLIB_MAX = value
}

func NewRand() int32 {
	for seed == prevSeed {
		seed = uint64(time.Now().UnixNano()*9457372 + 231453)
	}
	prevSeed = seed
	return int32(seed % RANDLIB_MAX)
}

func Randint(min, max int32) int32 {
	if min > max {
		max, min = min, max
	}
	var value int32 = min + (NewRand() % (max - min + 1))
	return value
}

func Randshort(min, max int16) int16 {
	if min > max {
		max, min = min, max
	}
	var value int16 = min + (int16(NewRand()) % (max - min + 1))
	return value
}

func Randlong(min, max int64) int64 {
	if min > max {
		max, min = min, max
	}
	var value int64 = min + (int64(NewRand()) % (max - min + 1))
	return value
}

func Randfloat(min, max float32) float32 {
	if min > max {
		min, max = max, min
	}
	var pre, post float32
	pre = float32(Randint(int32(min), int32(max-1)))
	post = float32(float64(NewRand()) / math.Pow(10, math.Log10(float64(RANDLIB_MAX))))
	return pre + post
}

func Randdouble(min, max float64) float64 {
	if min > max {
		min, max = max, min
	}
	var pre, post float64
	pre = float64(Randint(int32(min), int32(max-1)))
	post = float64(float64(NewRand()) / math.Pow(10, math.Log10(float64(RANDLIB_MAX))))
	return pre + post
}

func Randbool() bool {
	var value bool = false
	if Randint(0, 1) == 1 {
		value = true
	}
	return value
}

func Randchar(excludeWhitespaces bool) string {
	chars := []string{"0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "!", "\"", "#", "$", "%", "&", "'", "(", ")", "*", "+", ",", "-", ".", "/", ":", ";", "<", "=", ">", "?", "@", "[", "\\", "]", "^", "_", "`", "{", "|", "}", "~"}
	whitespaces := []string{" ", "\n", "\t"}
	length := int32(len(chars) - 1)
	var char string
	if excludeWhitespaces {
		char = chars[Randint(0, length)]
	} else {
		if Randint(0, 16) == 0 {
			char = whitespaces[Randint(0, 2)]
		} else {
			char = chars[Randint(0, length)]
		}

	}
	return char
}

func Randstring(excludeWhitespaces bool, length int32) string {
	var text string
	var char string
	for i := 0; i < int(length); i++ {
		char = Randchar(excludeWhitespaces)
		text += char
	}
	return text
}
